### PR TITLE
#35 add background music

### PR DESCRIPTION
**Pull request by onixred**
add background music
see detail  https://github.com/gpx-animator/gpx-animator/issues/35

### CODE REVIEW

File: src/main/java/app/gpx_animator/core/configuration/Configuration.java [hunk 1]
```diff
@@ -157,14 +162,14 @@ private Configuration(
...
-            final Double markerSize, final Font waypointFont, final Double waypointSize, final Double minLon, final Double maxLon,
+            final VideoCodec videoCodec, final MusicCodec musicCodec, final String attribution, final String information, final SpeedUnit speedUnit,
+            final Font font, final Double markerSize, final Font waypointFont, final Double waypointSize, final Double minLon, final Double maxLon,
             final Double minLat, final Double maxLat, final File logo, final Position logoPosition, final int logoMargin,
             final Position attributionPosition, final int attributionMargin,
             final Position informationPosition, final int informationMargin,
             final Position commentPosition, final int commentMargin,
-            final File photoDirectory, final long photoFreezeFrameTime, final Long photoTime, final Long photoAnimationDuration,
-            final boolean preview, final Long previewLength,
+            final File photoDirectory, final File inputMusic, final long photoFreezeFrameTime, final Long photoTime,
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/core/configuration/Configuration.java:171]: You added the parameter `File inputMusic` right between the parameters for the photos. There are already so many parameters on this constructor - I would like to keep similar parameters together, like not to break up the photo parameters. What about just switching the `photoDirectory` with the `inputMusic`? That would keep the photo parameters together. Or move the `inputMusic` right after the `musicCodec` to keep the music parameters togethers. I'm fine with both.

File: src/main/java/app/gpx_animator/core/configuration/Configuration.java [hunk 2]
```diff
@@ -562,14 +579,14 @@ public Configuration build() {
...
-                    preDrawTrack, keepFirstFrame, keepLastFrame, output, videoCodec, attribution, information,
+                    preDrawTrack, keepFirstFrame, keepLastFrame, output, videoCodec, musicCodec, attribution, information,
                     speedUnit, font, markerSize, waypointFont, waypointSize,
                     minLon, maxLon, minLat, maxLat,
                     logo, logoPosition, logoMargin,
                     attributionPosition, attributionMargin,
                     informationPosition, informationMargin,
                     commentPosition, commentMargin,
-                    photoDirectory, photoFreezeFrameTime, photoTime, photoAnimationDuration,
+                    photoDirectory, inputMusic, photoFreezeFrameTime, photoTime, photoAnimationDuration,
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/core/configuration/Configuration.java:589]: Same as with my comment above.

File: src/main/java/app/gpx_animator/core/data/MusicCodec.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2023 Contributors to the Sports-club
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/core/data/MusicCodec.java:2]: Please change this with the correct project name. We usually have the following line here: `Copyright Contributors to the GPX Animator project.`

**Review by McPringle — CHANGES_REQUESTED** [review]
Thank you very much for your contribution to GPX Animator, what a nice Xmas present! I have only a very easy to fix comment about the order of the constructor parameters. Merry Xmas and a Happy New Year!

File: src/main/java/app/gpx_animator/core/configuration/Configuration.java (hunk 1 above)
Comment (onixred) [reply src/main/java/app/gpx_animator/core/configuration/Configuration.java]: ok fixed

File: src/main/java/app/gpx_animator/core/configuration/Configuration.java (hunk 2 above)
Comment (onixred) [reply src/main/java/app/gpx_animator/core/configuration/Configuration.java]: reworked class

File: src/main/java/app/gpx_animator/core/data/MusicCodec.java (hunk 3 above)
Comment (onixred) [reply src/main/java/app/gpx_animator/core/data/MusicCodec.java]: changed copyright

**Comment by onixred** [comment]
I fixed

**Review by McPringle — APPROVED** [review]
Thank you very much for your contribution!

**Pull request merged**